From 2372ed3b73fa1b8bd203ce8bf61a78e3f5203395 Mon Sep 17 00:00:00 2001
From: Tejas Patel <tejas.patel1@einfochips.com>
Date: Fri, 12 Nov 2021 17:34:47 +0530
Subject: [PATCH 15/16] thor96: Update uart2 to work BT and uart debug together

This patch updates uart 2 node so that BT and uart debug can
work together

Change-Id: I327cdca999e614b1fa8c17f83f835295f50da4d8
Signed-off-by: Tejas Patel <tejas.patel1@einfochips.com>
Signed-off-by: Tanvi Chauhan <tanvi.chauhan@einfochips.com>

diff --git a/arch/arm64/boot/dts/freescale/imx8mq-thor96.dts b/arch/arm64/boot/dts/freescale/imx8mq-thor96.dts
index 098a525e6..a368463e7 100644
--- a/arch/arm64/boot/dts/freescale/imx8mq-thor96.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mq-thor96.dts
@@ -76,6 +76,13 @@
 		};
 	};
 
+	modem_reset: modem-reset {
+		compatible = "gpio-reset";
+		reset-gpios = <&gpio3 5 GPIO_ACTIVE_LOW>;
+		reset-delay-us = <2000>;
+		reset-post-delay-ms = <40>;
+		#reset-cells = <0>;
+	};
 
 	can_reset: can-reset {
 		compatible = "gpio-reset";
@@ -603,15 +610,7 @@
 	assigned-clock-parents = <&clk IMX8MQ_SYS1_PLL_80M>;
 	uart-has-rtscts;
 	status = "okay";
-
-	bluetooth {
-		compatible = "brcm,bcm43438-bt";
-		device-wakeup-gpios = <&gpio3 22 GPIO_ACTIVE_HIGH>;
-		host-wakeup-gpios = <&gpio3 14 GPIO_ACTIVE_HIGH>;
-		shutdown-gpios = <&gpio3 5 GPIO_ACTIVE_HIGH>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_bt_gpios>;
-	};
+	resets = <&modem_reset>;
 };
 
 /* LS-UART1 */
@@ -740,14 +739,6 @@
                 >;
         };
 
-	pinctrl_bt_gpios: btgpiosgrp {
-		fsl,pins = <
-			MX8MQ_IOMUXC_SAI5_RXD1_GPIO3_IO22		0x19
-			MX8MQ_IOMUXC_NAND_DQS_GPIO3_IO14		0x19
-			MX8MQ_IOMUXC_NAND_CLE_GPIO3_IO5			0x19
-		>;
-	};
-
 	pinctrl_csi1_pwn: csi1_pwn_grp {
 		fsl,pins = <
 			MX8MQ_IOMUXC_GPIO1_IO03_GPIO1_IO3		0x19
-- 
2.17.1

